### PR TITLE
Support args in cli

### PR DESCRIPTION
**Pull request by benyissa**
This is an example of usage:

```shell
oxo scan run --agent agent/ostorlab/nmap --arg ports:8080,8082 ip 8.8.8.8
```

image

### CODE REVIEW

File: src/ostorlab/cli/scan/run/run.py [hunk 1]
```diff
@@ -32,6 +35,15 @@
     help="List of agents keys. to use in the scan.",
     required=False,
 )
+@click.option(
+    "--arg",
+    multiple=True,
+    help="""Add an argument to an agent. The argument should be in the format: key:value where key is the name of the 
+    argument. the value is the value of the argument. Example: --arg ports=80,443,8080 --arg timeout=1000
```
Comment (rabsondev) [inline src/ostorlab/cli/scan/run/run.py:42]:
```suggestion
    argument. The value is the value of the argument. Example: --arg ports=80,443,8080 --arg timeout=1000
```

File: src/ostorlab/cli/scan/run/run.py [hunk 2]
```diff
@@ -139,7 +152,24 @@ def run(
...
+                                agent_setting.args.append(
+                                    utils_definitions.Arg(
+                                        name=current_arg.name,
+                                        type=supported_arg.get("type", "string"),
+                                        value=current_arg.value,
+                                    )
+                                )
+                except install_agent.AgentDetailsNotFound:
+                    console.warning(
+                        f"agent {agent_setting.key} not found. use oxo agent install {agent_setting.key} to install it"
```
Comment (rabsondev) [inline src/ostorlab/cli/scan/run/run.py:171]:
```suggestion
                        f"agent {agent_setting.key} not found. Use oxo agent install {agent_setting.key} to install it."
```

File: src/ostorlab/cli/types.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
+"""This module contains custom types for the CLI commands."""
+
+import logging
+import dataclasses
+from typing import Union
```
Comment (rabsondev) [inline src/ostorlab/cli/types.py:5]: Do you need this? Sometimes you're using `I`, other times `Union`.

File: src/ostorlab/cli/types.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+from ostorlab.cli import console as cli_console
+import click
```
Comment (rabsondev) [inline src/ostorlab/cli/types.py:8]:
```suggestion
import click

from ostorlab.cli import console as cli_console
```

File: src/ostorlab/cli/utils.py [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+        agent_details = response["data"]["agent"]
+        return agent_details
+
+
+def get_agent_definition(
+    agent_key: str,
+) -> agent_definitions.AgentDefinition:
+    """
+    Fetch args of an agent from container image.
+    """
```
Comment (rabsondev) [inline src/ostorlab/cli/utils.py:70]: Can you update the doctstring to include `agent_key` as well as what the function returns?

File: tests/cli/scan/run/run_test.py [hunk 6]
```diff
@@ -361,3 +361,22 @@ def testScanRunLocalRuntime_whenIInvalidYamlAgentGroupDefinition_showsErrorMessa
     assert isinstance(result.exception, SystemExit)
     assert result.exit_code == 2
     assert "Agent group definition YAML parse error" in result.output
+
+
+def testOstorlabScanRunCLI_whenWrongArgsFormatProvided_showsErrorMessage():
```
Comment (rabsondev) [inline tests/cli/scan/run/run_test.py:366]:
```suggestion
def testOstorlabScanRunCLI_whenWrongArgsFormatProvided_showsErrorMessage() -> None:
```

Can you also include a test for when the args format is correct?

**Comment by benyissa** [comment]
This pull request will be closed. and new PRs will be created. since some extra cases were discovered during the local testing of this feature. 

We will support string, number, and boolean first, and then a new implementation will take place for the array and object cases.

**Pull request closed without merging**